src/core/annotation.js

```diff
@@ -613,6 +613,40 @@ class Annotation {
     this.color = getRgbColor(color);
   }
 
+  /**
+   * Set the line endings; should only be used with specific annotation types.
+   * @param {Array} lineEndings - The line endings array.
+   */
+  setLineEndings(lineEndings) {
+    this.lineEndings = ["None", "None"]; // The default values.
+
+    if (Array.isArray(lineEndings) && lineEndings.length === 2) {
+      for (let i = 0, ii = lineEndings.length; i < ii; i++) {
+        const obj = lineEndings[i];
```

Comment:
Just a matter of taste, but it could be shorter with something like:
```js
if (obj instanceof Name && ["None", ...].includes(obj.name)) {
  this.lineEndings[i] = obj.name;
} else {
  warn(...);
}
```

Reply:
Unfortunately it doesn't actually get any shorter, it's less readable in my opinion (given the number of allowed values), and finally it'd also result in more iteration (since `includes` means iterating through the array); please see the code below.

Hence I'd like to keep the existing formatting, which I essentially "borrowed" from various getters in https://github.com/mozilla/pdf.js/blob/master/src/core/catalog.js

```javascript
if (
  obj instanceof Name &&
  [
    "None",
    "Square",
    "Circle",
    "Diamond",
    "OpenArrow",
    "ClosedArrow",
    "Butt",
    "ROpenArrow",
    "RClosedArrow",
    "Slash",
  ].includes(obj.name)
) {
  this.lineEndings[i] = obj.name;
} else {
  warn(`Ignoring invalid lineEnding: ${obj}`);
}
```